maidsafe/safe-build-infrastructure
# Prepare Jenkins for DNS Record

Hi Stephen,

As discussed, this contains some work to prepare us for using jenkins.maidsafe.net:

* Runs Jenkins on port 80 for both dev and prod environments
* Assigns a pre-allocated Elastic IP to the Jenkins master instance
* Allows the provision of the dev and prod environments at the same time

To test this:

* Create the dev environment with `make env-jenkins-dev-aws`
* Create the production environment as per the instructions in the README (still the same)

*Note*: when provisioning the Windows slave, if you see any errors with Chocolatey packages, just run the provision again. The packages that most typically cause this are `jre8` and `mingw`.

Acceptance criteria:
* No errors provisioning either of the environments (other than intermittent failures with Chocolatey packages)
* Access the dev environment with the DNS name of the master without having to specify port 8080
* Access the prod environment with the DNS name of the master without having to specify port 8080
* Access the prod environment with the Elastic IP (3.8.245.27)

Cheers,

Chris
